linkcheck/checker/fileurl.py

```diff
@@ -234,8 +234,7 @@ def read_content(self):
         with links to the files."""
         if self.is_directory():
             data = get_index_html(get_files(self.get_os_filename()))
-            if isinstance(data, str):
-                data = data.encode("iso8859-1", "ignore")
+            data = data.encode("iso8859-1", "ignore")
```

Comment:
This is a correct refactoring, but it brings attention to buggy code.  ISO-8859-1 is not the right choice for arbitrary filenames.  UTF-8 might work better.  Ideally we'd plumb things up so that get_content() that calls get_raw_content() that calls download_content() that calls read_content() wouldn't need to have BeautifulSoup guess what the encoding is, given that it's hardcoded right here!

(Maybe get_index_html() ought to generate a `<meta charset="UTF-8">`.)

In case it wasn't clear: this change here is fine, and the bugfix belongs to a separate PR.

Reply:
Raised issue #451.